Jenkinsfile

```diff
@@ -57,25 +72,32 @@ node ("ts-engine && heavy && java8") {
         }
     }
 
-    stage('Integration Tests') {
-        try {
-            sh './gradlew --console=plain integrationTest'
-        } finally {
-            junit testResults: '**/build/test-results/integrationTest/*.xml', allowEmptyResults: true
-        }
-    }
-
     stage('Analytics') {
-        sh './gradlew --console=plain check -x test spotbugsmain'
-        recordIssues tool: checkStyle(pattern: '**/build/reports/checkstyle/*.xml')
-        recordIssues tool: spotBugs(pattern: '**/build/reports/spotbugs/main/*.xml', useRankAsPriority: true)
-        recordIssues tool: pmdParser(pattern: '**/build/reports/pmd/*.xml')
-        recordIssues tool: taskScanner(includePattern: '**/*.java,**/*.groovy,**/*.gradle', lowTags: 'WIBNIF', normalTags: 'TODO', highTags: 'ASAP')
+        sh './gradlew --console=plain check -x test'
+        // the default resolution when omitting `defaultBranch` is to `master` - which is wrong in our case. 
+        discoverGitReferenceBuild(defaultBranch: 'develop') //TODO: does this also work for PRs with different base branch?
+        recordIssues skipBlames: true, qualityGates: [[threshold: 1, type: 'NEW', unstable: true]], 
+            tools: [
+                checkStyle(pattern: '**/build/reports/checkstyle/*.xml'),
+                spotBugs(pattern: '**/build/reports/spotbugs/main/*.xml', useRankAsPriority: true),
+                pmdParser(pattern: '**/build/reports/pmd/*.xml')
+            ] 
+            
+        recordIssues skipBlames: true, 
+            tool: taskScanner(includePattern: '**/*.java,**/*.groovy,**/*.gradle', lowTags: 'WIBNIF', normalTags: 'TODO', highTags: 'ASAP')
     }
 
     stage('Documentation') {
         sh './gradlew --console=plain javadoc'
         step([$class: 'JavadocArchiver', javadocDir: 'engine/build/docs/javadoc', keepAll: false])
-        recordIssues tool: javaDoc()
+        recordIssues skipBlames: true, tool: javaDoc()
+    }
+
+    stage('Integration Tests') {
```

Comment:
I've moved integration tests down as they are long running, and it might be desirable to get fast feedback on the other analytic reports. 

I still think that **Publish** and **Documentation** should be the last stages in the pipeline, but since we decided to publish early and run checks only later I figured that putting the short-running stages first makes sense.